chia/farmer/farmer_api.py

```diff
@@ -354,8 +363,11 @@ async def new_signage_point(self, new_signage_point: farmer_protocol.NewSignageP
 
             msg = make_msg(ProtocolMessageTypes.new_signage_point_harvester, message)
             await self.farmer.server.send_to_all([msg], NodeType.HARVESTER)
```

Comment:
How about simply adding it to the list right here, and then not having to worry about an exception?
I'm not too worried about something getting added twice as the processing should be pretty fast, if you disagree let me know.

Reply:
Duplicate SP processing by the harvesters within a very short timeframe will cause multiple reads and processing in case of compressed plots, which is undesired

Imho ensuring we do not do unnecessary reads and compute is important, is it not?

Reply:
fair point with the compressed plots.